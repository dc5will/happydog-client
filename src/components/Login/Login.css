.loginForm {
    margin-bottom: 30px;
}

.loginButton {
    margin-top: 20px;
    font-size: 24px;
    background-color: rgb(192, 81, 81);
}

.loginButton:hover {
    background-color: rgb(139, 36, 36);
}

.loginButton:active {
    background-color: rgb(138, 32, 32);
    box-shadow: 0 5px #666;
    transform: translateY(2px);
}



label {
    font-size: 20px;
}

h3 {
    margin: 40px;
}