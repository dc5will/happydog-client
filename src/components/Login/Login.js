import React, { Component } from "react";
// import { Link } from 'react-router-dom'
import AuthApiService from "../../services/auth-api-service";
import TokenService from "../../services/token-service";
import NotesContext from "../../contexts/NotesContext";
import Header from "../../components/Header/Header";
import { Input, Button, Section } from "../../components/Utils/Utils";
import Footer from '../../components/Footer/Footer'
import './Login.css'

class Login extends Component {
  static defaultProps = {
    onLoginSuccess: () => {}
  };
  static contextType = NotesContext;

  state = { error: null };

  handleSubmitJwtAuth = ev => {
    ev.preventDefault();
    this.setState({ error: null });
    const { user_name, password } = ev.target;

    AuthApiService.postLogin({
      user_name: user_name.value,
      password: password.value
    })
      .then(res => {
        user_name.value = "";
        password.value = "";
        TokenService.saveAuthToken(res.authToken);
        this.props.history.push("/my-notes");
        this.context.getNotes();
      })
      .catch(res => {
        this.setState({ error: res.error });
      });
  };

  render() {
    const { error } = this.state;

    return (
      <>
        <Header />
        <Section className="LoginPage">
          <h2>Welcome back!</h2>
          <h3>Please log in below:</h3>
          <form className="loginForm" onSubmit={this.handleSubmitJwtAuth}>
            <div role="alert">{error && <p className="red">{error}</p>}</div>
            <div className="user_name">
              <label htmlFor="LoginForm__user_name">User name</label>
              <Input required name="user_name" id="LoginForm__user_name" />
            </div>
            <div className="password">
              <label htmlFor="LoginForm__password">Password</label>
              <Input
                required
                name="password"
                type="password"
                id="LoginForm__password"
              />
            </div>
            <Button type="submit">Login</Button>
          </form>
        </Section>
        <Footer />
      </>
    );
  }
}

export default Login;
