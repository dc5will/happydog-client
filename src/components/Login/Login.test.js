import React from "react";
import { shallow } from "enzyme";
import Login from "./Login";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

describe(`LandingPage component`, () => {
  it("renders without crashing", () => {
    shallow(<Login />);
  });

  it("renders the UI as expected", () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Login />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
