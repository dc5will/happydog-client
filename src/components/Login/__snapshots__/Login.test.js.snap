// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage component renders the UI as expected 1`] = `
Array [
  <nav
    className="Header"
  >
    <h1>
      <a
        href="/my-notes"
        onClick={[Function]}
      >
        Happy dog
      </a>
    </h1>
    <div
      className="Header_not-logged-in"
    >
      <a
        href="/login"
        onClick={[Function]}
      >
        Log in
      </a>
      <a
        href="/register"
        onClick={[Function]}
      >
        Register
      </a>
    </div>
  </nav>,
  <section
    className={
      Array [
        "Section",
        undefined,
        "LoginPage",
      ]
    }
  >
    <h2>
      Welcome back!
    </h2>
    <h3>
      Please log in below:
    </h3>
    <form
      className="loginForm"
      onSubmit={[Function]}
    >
      <div
        role="alert"
      />
      <div
        className="user_name"
      >
        <label
          htmlFor="LoginForm__user_name"
        >
          User name
        </label>
        <input
          className="Input "
          id="LoginForm__user_name"
          name="user_name"
          required={true}
        />
      </div>
      <div
        className="password"
      >
        <label
          htmlFor="LoginForm__password"
        >
          Password
        </label>
        <input
          className="Input "
          id="LoginForm__password"
          name="password"
          required={true}
          type="password"
        />
      </div>
      <button
        className="Button "
        type="submit"
      >
        Login
      </button>
    </form>
  </section>,
  <footer
    className="footer"
  >
    <div
      className="footer_brand-container"
    />
  </footer>,
]
`;
